### PR TITLE
Static Frames Validation Security Fix

### PR DESCRIPTION
**Pull request by codabrink**
## Summary by CodeRabbit

- **New Features**
	- Enhanced validation logic for frame submissions, focusing on authorization checks.
	- Reintroduced methods for checking address and installation authorization in the client.

- **Bug Fixes**
	- Improved error handling for unauthorized installations and wallet addresses.

- **Chores**
	- Updated package dependencies for `@xmtp/frames-validator` and `@xmtp/frames-client` to the latest version. 

- **Tests**
	- Updated test cases to reflect new endpoint identifiers for improved accuracy.

### CODE REVIEW

File: packages/frames-validator/src/validation.ts [hunk 1]
```diff
@@ -48,9 +48,12 @@ export async function validateFramesPost(
     }
   } else {
     // make sure inbox IDs match
-    const addressInboxId = await getInboxIdForAddress(walletAddress, env);
-    if (inboxId !== addressInboxId) {
-      throw new Error("Invalid inbox ID");
+    const authorized = await Client.isInstallationAuthorized(
```
Comment (neekolas) [inline packages/frames-validator/src/validation.ts:51]: This is nice and tidy

File: sdks/node-sdk/package.json [hunk 2]
```diff
@@ -53,7 +53,7 @@
     "@xmtp/content-type-group-updated": "^1.0.1",
     "@xmtp/content-type-primitives": "^1.0.3",
     "@xmtp/content-type-text": "^1.0.1",
-    "@xmtp/node-bindings": "^0.0.22",
```
Comment (insipx) [inline sdks/node-sdk/package.json:56]: if you bump to `0.0.25` you can get the release into here I just pushed too, to fix streams

Comment (codabrink) [reply sdks/node-sdk/package.json:56]: Sure thing

File: packages/frames-validator/package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
   "dependencies": {
     "@noble/curves": "^1.3.0",
     "@noble/hashes": "^1.4.0",
-    "@xmtp/node-sdk": "^0.0.27",
+    "@xmtp/node-sdk": "^0.0.29",
```
Comment (nplasterer) [inline packages/frames-validator/package.json:43]: I think you will also need to update line 3 of this file to make a new validator release

**Review by nplasterer — APPROVED** [review]

File: packages/frames-validator/package.json (hunk 3 above)
Comment (codabrink) [reply packages/frames-validator/package.json:43]: Ah yeah, good point

**Review by neekolas — APPROVED** [review]

File: packages/frames-validator/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@xmtp/frames-validator",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (rygine) [inline packages/frames-validator/package.json:3]: version updates should be done with a changeset

File: sdks/node-sdk/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@xmtp/node-sdk",
-  "version": "0.0.28",
+  "version": "0.0.29",
```
Comment (rygine) [inline sdks/node-sdk/package.json:3]: see other comment on version updates

File: packages/frames-validator/package.json (hunk 4 above)
Comment (nplasterer) [reply packages/frames-validator/package.json:3]: Ahh I think this was my fault. It wasn't super clear to me from reading the readmes how to do releases.

**Pull request merged**